clustered/client/src/test/java/org/ehcache/clustered/client/docs/SampleLoaderWriter.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.docs;
+
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Sample Loader Writer used for user documentation
+ */
+public class SampleLoaderWriter<K, V> implements CacheLoaderWriter<K, V> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SampleLoaderWriter.class);
+
+  private final Map<K, V> data = new HashMap<>();
+  private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+
+  public SampleLoaderWriter(Map<K, V> initialData) {
+    data.putAll(initialData);
+  }
+
+  @Override
+  public V load(K key) {
+    lock.readLock().lock();
+    try {
+      V value = data.get(key);
+      LOGGER.info("Key - '{}', Value - '{}' successfully loaded", key, value);
+      return value;
+    } finally {
+      lock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public Map<K, V> loadAll(Iterable<? extends K> keys) {
+    throw new UnsupportedOperationException("Implement me!");
```

Comment:
Don't the bulk methods have default implementations. Why are we overriding them?

Reply:
Same as above.